### ISSUE TITLE
Paywall Template 7 (Siamese) not recognized (react-native-purchases-ui 7.28.1)

### ISSUE THREAD

**Issue by Tom-Standen** [RevenueCat/react-native-purchases#1044]
- [x] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/react-native-purchases/blob/main/CONTRIBUTING.md)
- [x] I have searched the [Community](https://community.revenuecat.com)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/react-native-purchases/issues)

**Describe the bug**
I have moved to using Paywall 7 - Siamese from Paywall 5 - Bengal as it is the perfect use for our application. However, when I run the same code as before but with the new paywall I see an error that makes it look like the sdk isn't prepared for 7 - Siamese yet. 

However, I am receiving the following error:

ERROR TEXT:
Template not recognized: 7.
You can fix this by editing the paywall in the RevenueCat dashboard
The displayed paywall contains default configuration
This error will be hidden in production.

1. Environment
   1. Platform: React Native with Expo managed
   2. SDK version: 7.28.1
   3. OS version: MacOS Sonoma 14.5
   4. Xcode/Android Studio version:
   5. React Native version: 0.74.3 
   6. SDK installation (CocoaPods + version or manual): Via Expo 
   7. How widespread is the issue. Percentage of devices affected.
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue
3. Steps to reproduce, with a description of expected vs. actual behavior
4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)

**Additional context**
Here is a screenshot of the error:
![IMG_07B94B658520-1](https://github.com/user-attachments/assets/105f0faa-bee1-41d3-a939-ceb217bad08a)

Here is the function that is causing the error:
```
export const presentPaywallIfNeeded = async (entitlementIdString, offering = null) => {
    const premiumEnabled = getRemoteConfigFeatureFlags()?.premium_enabled ?? false;
    if (!premiumEnabled) return true;

    console.log("DEBUG ** presentPaywallIfNeeded()", entitlementIdString, offering);
    const paywallOptions = {
        requiredEntitlementIdentifier: entitlementIdString,
        displayCloseButton: true,
        offering: offering,
        onPurchaseStarted: () => {
            console.log("Purchase started");
        },
        onPurchaseCompleted: ({ customerInfo }) => {
            console.log("Purchase completed:", customerInfo);
            updateSubscriptionState();
        },
        onPurchaseError: (error) => {
            console.error("Purchase error:", error);
        },
        onRestoreStarted: () => {
            console.log("Restore started");
        },
        onRestoreCompleted: ({ customerInfo }) => {
            console.log("Restore completed:", customerInfo);
            updateSubscriptionState();
        },
        onRestoreError: (error) => {
            console.error("Restore error:", error);
        },
        onDismiss: () => {
            console.log("Paywall dismissed");
        }
    };

    // Add offering to paywallOptions if it's provided
    if (offering !== null) {
        console.log("DEBUG ** adding offering to paywallOptions")
        paywallOptions.offering = offering;
    }
    console.log("DEBUG ** paywallOptions:", paywallOptions);

    const paywallResult = await RevenueCatUI.presentPaywallIfNeeded(paywallOptions);

    console.log("paywallResult:", paywallResult);
    switch (paywallResult) {
        case PAYWALL_RESULT.ERROR:
        case PAYWALL_RESULT.CANCELLED:
            return false;
        case PAYWALL_RESULT.NOT_PRESENTED:
        case PAYWALL_RESULT.PURCHASED:
        case PAYWALL_RESULT.RESTORED:
            return true;
        default:
            return false;
    }
}
```

**Comment by mshmoustafa** [comment]
Hi @Tom-Standen,
I think you have a couple of support tickets open about this but this error is due to the fact that this specific template has not yet been implemented for our React Native SDK. We plan on implementing it soon, on the next major version update. If you watch the releases page you'll be notified of every new release so you can update as soon as it's available.

**Comment by Tom-Standen** [comment]
Hey @mshmoustafa, thanks for the heads up. My mistake for sounding the alarm, I thought it was a bug!

Best,
Tom

**Comment by mshmoustafa** [comment]
No worries, you wouldn't have known without asking!

**Comment by geoffcfchen** [comment]
I got exactly the same issue. Hopefully it will be fixed soon.
![IMG_4559](https://github.com/user-attachments/assets/c28723f7-8310-487e-8804-deaadb00f835)

**Comment by mshmoustafa** [comment]
@geoffcfchen what version of our SDK are you using? You should be on 8.0.1

**Comment by geoffcfchen** [comment]
Hi, @mshmoustafa 

After updating to 8.0.1, ios version does work. However, android does not work. 

when I use the following code to print out offerings.all

```
  useEffect(() => {
    setLoading(true);
    const fetchOfferings = async () => {
      // console.log("test");
      try {
        const offerings = await Purchases.getOfferings();
        if (Platform.OS === "android") {
          console.log("android offerings", offerings.all);
        }
        if (Platform.OS === "ios") {
          console.log("ios offerings", offerings.all);
        }

        if (offerings) {
          // console.log("offering", offerings.all["onboarding"]);
          if (Platform.OS === "ios") {
            setOffering(offerings.all["Standard"]);
          } else {
            setOffering(offerings.all["Standard"]);
          }
        }
      } catch (e) {
        console.error(e);
      } finally {
        setLoading(false);
      }
    };

    fetchOfferings();
  }, []);
```
  
  for ios, I got 
  
```
  {
  "Default": {
    "annual": { "identifier": "$rc_annual", "offeringIdentifier": "Default", "packageType": "ANNUAL", "presentedOfferingContext": [Object], "product": [Object] },
    "availablePackages": [[Object], [Object]],
    "identifier": "Default",
    "metadata": {},
    "monthly": { "identifier": "$rc_monthly", "offeringIdentifier": "Default", "packageType": "MONTHLY", "presentedOfferingContext": [Object], "product": [Object] },
    "serverDescription": "The standard set of products"
  },
  "Standard": {
    "availablePackages": [[Object], [Object], [Object], [Object]],
    "identifier": "Standard",
    "metadata": {},
    "serverDescription": "Standard package"
  },
  "onboarding": {
    "annual": { "identifier": "$rc_annual", "offeringIdentifier": "onboarding", "packageType": "ANNUAL", "presentedOfferingContext": [Object], "product": [Object] },
    "availablePackages": [[Object]],
    "identifier": "onboarding",
    "metadata": {},
    "serverDescription": "Offering for the onboarding flow"
  }
}
```

But for android, I got 

```
{
  "Default": {
    "annual": { "identifier": "$rc_annual", "offeringIdentifier": "Default", "packageType": "ANNUAL", "presentedOfferingContext": [Object], "product": [Object] },
    "availablePackages": [[Object], [Object]],
    "identifier": "Default",
    "lifetime": null,
    "metadata": {},
    "monthly": { "identifier": "$rc_monthly", "offeringIdentifier": "Default", "packageType": "MONTHLY", "presentedOfferingContext": [Object], "product": [Object] },
    "serverDescription": "The standard set of products",
    "sixMonth": null,
    "threeMonth": null,
    "twoMonth": null,
    "weekly": null
  },
  "onboarding": {
    "annual": { "identifier": "$rc_annual", "offeringIdentifier": "onboarding", "packageType": "ANNUAL", "presentedOfferingContext": [Object], "product": [Object] },
    "availablePackages": [[Object]],
    "identifier": "onboarding",
    "lifetime": null,
    "metadata": {},
    "monthly": null,
    "serverDescription": "Offering for the onboarding flow",
    "sixMonth": null,
    "threeMonth": null,
    "twoMonth": null,
    "weekly": null
  }
}
```


You can see obviously "Standard" is missing in android, which I use it for template 7.

**Comment by mshmoustafa** [comment]
@geoffcfchen I think this issue is separate from paywalls, from what I understand getOfferings doesn't depend on your paywall template. Can you run this curl request with the same app user ID and API key that you use in your code and let me know what the response is?

```
curl -X GET \
  https://api.revenuecat.com/v1/subscribers/{app_user_id}/offerings \
  -H "Authorization: Bearer YOUR_API_KEY" \
  -H "X-Platform: android"
```

**Comment by geoffcfchen** [comment]
Hi @mshmoustafa 

Thank you for helping out. here is the output

`{"current_offering_id":"Standard","offerings":[{"description":"The standard set of products","identifier":"Default","metadata":null,"packages":[{"identifier":"$rc_monthly","platform_product_identifier":"rc_premium_900_1m"},{"identifier":"$rc_annual","platform_product_identifier":"rc_premium_9399_1y"}],"paywall":{"asset_base_url":"https://assets.pawwalls.com","config":{"colors":{"dark":{"accent_1":"#24D8AD","accent_2":"#24D8AD","accent_3":"#606463","background":"#000000","call_to_action_background":"#24D8AD","call_to_action_foreground":"#FFFFFF","text_1":"#FFFFFF","text_2":"#000000","text_3":"#FFFFFF"},"light":{"accent_1":"#008575","accent_2":"#008575","accent_3":"#DFDFDF","background":"#43FDD8","call_to_action_background":"#2A2A2A","call_to_action_foreground":"#FFFFFF","text_1":"#000000","text_2":"#FFFFFF","text_3":"#000000"}},"default_package":"$rc_annual","display_restore_purchases":true,"images":{"header":"1138866_1723875881.png"},"images_heic":{"header":"1138866_1723875881.heic"},"images_webp":{"header":"1138866_1723875881.webp"},"packages":["$rc_monthly","$rc_annual"],"privacy_url":"https://vetcation.com/#/privacy-policy","tos_url":"https://vetcation.com/#/SMSTerms"},"default_locale":null,"localized_strings":{"en_US":{"call_to_action":"Support the Community Today","features":[{"icon_id":"tick","title":"Post unlimited questions"},{"icon_id":"tick","title":"Support the Community"},{"icon_id":"tick","title":"Earn a premium badge"}],"offer_details":"Learn from the experts for just {{ total_price_and_per_month }}","offer_name":"{{ sub_period }}","title":"Learn from the Veterinary Expert"}},"localized_strings_by_tier":{},"revision":1,"template_name":"5","zero_decimal_place_countries":{"apple":["TWN","KAZ","MEX","PHL","THA"],"google":["TW","KZ","MX","PH","TH"]}}},{"description":"Offering for the onboarding flow","identifier":"onboarding","metadata":null,"packages":[{"identifier":"$rc_annual","platform_product_identifier":"rc_11499_1y"}]},{"description":"Standard package","identifier":"Standard","metadata":null,"packages":[{"identifier":"Basic Monthly","platform_product_identifier":"rc_basic_400_1m"},{"identifier":"Premium Monthly","platform_product_identifier":"rc_premium_900_1m"},{"identifier":"Basic Yearly","platform_product_identifier":"rc_basic_4200_1y"},{"identifier":"Premium Yearly","platform_product_identifier":"rc_premium_9399_1y"}],"paywall":{"asset_base_url":"https://assets.pawwalls.com","config":{"colors":{"dark":{"accent_1":"#24D8AD","accent_2":"#24D8AD","accent_3":"#606463","background":"#000000","call_to_action_background":"#24D8AD","call_to_action_foreground":"#FFFFFF","text_1":"#FFFFFF","text_2":"#000000","text_3":"#FFFFFF","tier_control_background":"#606463","tier_control_foreground":"#000000","tier_control_selected_background":"#008575","tier_control_selected_foreground":"#FFFFFF"},"light":{"accent_1":"#30A0F8","accent_2":"#30A0F8","accent_3":"#DFDFDF","background":"#FFFFFF","call_to_action_background":"#A8EE76","call_to_action_foreground":"#000000","text_1":"#000000","text_2":"#FFFFFF","text_3":"#000000","tier_control_background":"#EFEFF0","tier_control_foreground":"#000000","tier_control_selected_background":"#A8EE76","tier_control_selected_foreground":"#000000"}},"colors_by_tier":{"68w4WLNkAL":{"dark":{"accent_1":"#24D8AD","accent_2":"#24D8AD","accent_3":"#606463","background":"#000000","call_to_action_background":"#24D8AD","call_to_action_foreground":"#FFFFFF","text_1":"#FFFFFF","text_2":"#000000","text_3":"#FFFFFF","tier_control_background":"#606463","tier_control_foreground":"#000000","tier_control_selected_background":"#008575","tier_control_selected_foreground":"#FFFFFF"},"light":{"accent_1":"#30A0F8","accent_2":"#30A0F8","accent_3":"#DFDFDF","background":"#43fdd8","call_to_action_background":"#2a2a2a","call_to_action_foreground":"#ffffff","text_1":"#000000","text_2":"#FFFFFF","text_3":"#000000","tier_control_background":"#EFEFF0","tier_control_foreground":"#000000","tier_control_selected_background":"#2a2a2a","tier_control_selected_foreground":"#ffffff"}},"zDnM0sI2PS":{"dark":{"accent_1":"#24D8AD","accent_2":"#24D8AD","accent_3":"#606463","background":"#000000","call_to_action_background":"#24D8AD","call_to_action_foreground":"#FFFFFF","text_1":"#FFFFFF","text_2":"#000000","text_3":"#FFFFFF","tier_control_background":"#606463","tier_control_foreground":"#000000","tier_control_selected_background":"#008575","tier_control_selected_foreground":"#FFFFFF"},"light":{"accent_1":"#30A0F8","accent_2":"#30A0F8","accent_3":"#DFDFDF","background":"#43fdd8","call_to_action_background":"#2a2a2a","call_to_action_foreground":"#ffffff","close_button":"#2A2A2A","text_1":"#000000","text_2":"#FFFFFF","text_3":"#000000","tier_control_background":"#EFEFF0","tier_control_foreground":"#000000","tier_control_selected_background":"#2a2a2a","tier_control_selected_foreground":"#ffffff"}}},"display_restore_purchases":true,"images":{"header":null},"images_by_tier":{"68w4WLNkAL":{"header":"1138866_1722750780.png"},"zDnM0sI2PS":{"header":"1138866_1722750780.png"}},"images_heic":{"header":null},"images_heic_by_tier":{"68w4WLNkAL":{"header":"1138866_1722750780.heic"},"zDnM0sI2PS":{"header":"1138866_1722750780.heic"}},"images_webp":{"header":null},"images_webp_by_tier":{"68w4WLNkAL":{"header":"1138866_1722750780.webp"},"zDnM0sI2PS":{"header":"1138866_1722750780.webp"}},"packages":[],"privacy_url":"https://vetcation.com/#/privacy-policy","tiers":[{"default_package":"Basic Yearly","id":"zDnM0sI2PS","packages":["Basic Monthly","Basic Yearly"]},{"default_package":"Basic Yearly","id":"68w4WLNkAL","packages":["Premium Monthly","Premium Yearly"]}],"tos_url":"https://vetcation.com/#/SMSTerms"},"default_locale":null,"localized_strings":{},"localized_strings_by_tier":{"en_US":{"68w4WLNkAL":{"call_to_action":"Support the Community Today","features":[{"icon_id":"tick","title":"3x visibility for posted questions"},{"icon_id":"tick","title":"2x Support the Community"},{"icon_id":"stars","title":"Earn a premium badge"}],"offer_details":"Join the community for just {{ price_per_period }}","offer_name":"{{ sub_period }}","tier_name":"Premium","title":"Learn from the Veterinary Expert"},"zDnM0sI2PS":{"call_to_action":"Support the Community Today","features":[{"icon_id":"tick","title":"Post unlimited questions"},{"icon_id":"tick","title":"Support the Community"}],"offer_details":"Join the community for just {{ price_per_period }}","offer_name":"{{ sub_period }}","tier_name":"Basic","title":"Learn from the Veterinary Expert"}}},"revision":2,"template_name":"7","zero_decimal_place_countries":{"apple":["TWN","KAZ","MEX","PHL","THA"],"google":["TW","KZ","MX","PH","TH"]}}}],"placements":{"fallback_offering_id":"Standard"}}`

**Comment by geoffcfchen** [comment]
Hi @mshmoustafa 

I found that Android does return correct offerings without missing any object today. Maybe you guys fix it? thanks.

for ios, I got
```
{
  "Default": {
    "annual": {
      "identifier": "$rc_annual",
      "offeringIdentifier": "Default",
      "packageType": "ANNUAL",
      "presentedOfferingContext": [Object],
      "product": [Object]
    },
    "availablePackages": [
      [Object],
      [Object]
    ],
    "identifier": "Default",
    "metadata": {},
    "monthly": {
      "identifier": "$rc_monthly",
      "offeringIdentifier": "Default",
      "packageType": "MONTHLY",
      "presentedOfferingContext": [Object],
      "product": [Object]
    },
    "serverDescription": "The standard set of products"
  },
  "Standard": {
    "availablePackages": [
      [Object],
      [Object],
      [Object],
      [Object]
    ],
    "identifier": "Standard",
    "metadata": {},
    "serverDescription": "Standard package"
  },
  "onboarding": {
    "annual": {
      "identifier": "$rc_annual",
      "offeringIdentifier": "onboarding",
      "packageType": "ANNUAL",
      "presentedOfferingContext": [Object],
      "product": [Object]
    },
    "availablePackages": [
      [Object]
    ],
    "identifier": "onboarding",
    "metadata": {},
    "serverDescription": "Offering for the onboarding flow"
  }
}

```
For android, NOW I got

```
{
  "Default": {
    "annual": {
      "identifier": "$rc_annual",
      "offeringIdentifier": "Default",
      "packageType": "ANNUAL",
      "presentedOfferingContext": [Object],
      "product": [Object]
    },
    "availablePackages": [
      [Object],
      [Object]
    ],
    "identifier": "Default",
    "lifetime": null,
    "metadata": {},
    "monthly": {
      "identifier": "$rc_monthly",
      "offeringIdentifier": "Default",
      "packageType": "MONTHLY",
      "presentedOfferingContext": [Object],
      "product": [Object]
    },
    "serverDescription": "The standard set of products",
    "sixMonth": null,
    "threeMonth": null,
    "twoMonth": null,
    "weekly": null
  },
  "Standard": {
    "annual": null,
    "availablePackages": [
      [Object],
      [Object],
      [Object],
      [Object]
    ],
    "identifier": "Standard",
    "lifetime": null,
    "metadata": {},
    "monthly": null,
    "serverDescription": "Standard package",
    "sixMonth": null,
    "threeMonth": null,
    "twoMonth": null,
    "weekly": null
  },
  "onboarding": {
    "annual": {
      "identifier": "$rc_annual",
      "offeringIdentifier": "onboarding",
      "packageType": "ANNUAL",
      "presentedOfferingContext": [Object],
      "product": [Object]
    },
    "availablePackages": [
      [Object]
    ],
    "identifier": "onboarding",
    "lifetime": null,
    "metadata": {},
    "monthly": null,
    "serverDescription": "Offering for the onboarding flow",
    "sixMonth": null,
    "threeMonth": null,
    "twoMonth": null,
    "weekly": null
  }
}

```

**Comment by mshmoustafa** [comment]
@geoffcfchen we didn't fix anything related to this so I think it was just a broken cache, sandbox glitches, etc. It's common for that to happen in testing.
I'll ask the mobile engineering team to look into that crash and report back what they say

**Comment by joshdholtz** [comment]
@geoffcfchen Hey! The multi-tier paywall isn't available until React Native version `8.0.0`. Sorry about that! We will make it more clear in the dashboard and docs what SDK versions paywalls are available on.

**Comment by joshdholtz** [comment]
> @joshdholtz, Hey. Thanks for the information. Just to note that I am using "react-native-purchases": "8.0.1", "react-native-purchases-ui": "8.0.1" and although template 7 works on iOS, it does not work on Android (see the above crash log).

Ah, sorry! Coming back from being sick this weekend so not fully in it yet 🫠 Looking hard this time and will come back with something actually useful 😇

**Comment by joshdholtz** [comment]
@geoffcfchen Okay, now after _really_ looking at the crash, it seems like you might not have the correct version of `androidx.compose.animation` 🤔 Template 7 uses `androidx.compose.animation.AnimatedContent` but its not able to find that...

Do you happen to know if you have this dependency pinned at a lower version? This should be found in Jetpack Compose version `1.3.0` and up

**Comment by geoffcfchen** [comment]
@joshdholtz, we are using Expo Managed Workflow right now, but maybe I can find a way to implement your suggestions. Let me try it and report back...

**Comment by geoffcfchen** [comment]
@joshdholtz Again, thanks for looking into this problem. Would you mind to let me know if you have any updates on this? or any advice on this? Thanks.

**Comment by Tom-Standen** [comment]
Hey @joshdholtz @mshmoustafa 

Thought I would use this thread instead of creating a new issue as it's related.

Great job on Siamese - it's perfect for our use case so thank you!

One quick issue though. We seem to be seeing the package id come through on our template instead of the dynamic '{{ sub_period }} Plan' that we were hoping for. 

I assumed that this was due to TestFlight or dev bugs but we are now in **production** and seeing this still on our PayWall. In the short term I will recreate all packages and give them ids of exactly how I want them to appear int the app but thought you guys would want to know.

**Production PayWall**

![IMG_1153](https://github.com/user-attachments/assets/95399e7f-b7d6-489d-b989-a83502df4c26)

RevCat Console 1

![image](https://github.com/user-attachments/assets/f5fdd0cd-11ff-41e8-b53e-4e0be4018a69)

RevCat Packages

![image](https://github.com/user-attachments/assets/0e30f31e-842f-41bc-a2c0-d629f056984c)

**Comment by mshmoustafa** [comment]
@Tom-Standen let me check on that

**Comment by mshmoustafa** [comment]
@Tom-Standen just letting you know that the engineering team is looking into it!

**Comment by mshmoustafa** [comment]
@Tom-Standen a fix is out here! https://github.com/RevenueCat/purchases-ios/pull/4273. It is available in v8.2.3 of the React Native SDK: https://github.com/RevenueCat/react-native-purchases/releases/tag/8.2.3
